### DISCUSSION TITLE
How to change output of mediaMTX from WebRTC to something else that support MJPEG

### DISCUSSION THREAD

**Discussion by nqa141019** [bluenviron/mediamtx#3782] [Questions]
### Question

Hi, I was configuration mediaMTX on hailo15 to pass rtsp video stream ( h264, h265 ) into mjpeg to show on web but still failed so I want to ask for help. Do I need to change anything in mediamtx.yml or can I do it by any others way? Thank you

**Comment by aler9** [comment]
Hello, in order to show streams on the web you don't need M-JPEG but you can use HLS or WebRTC, that are natively compatible with H264 and H265.

HLS:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#hls

WebRTC:

https://github.com/bluenviron/mediamtx?tab=readme-ov-file#webrtc

If you still want to use M-JPEG, you can use FFmpeg to pull the video stream from the server and re-encode it from H264 to M-JPEG, serving the resulting file through a static HTTP server.

**Reply by nqa141019** [reply]
thank you Alessandro
